### DISCUSSION TITLE
WLAN Pi OS v3.1.1 button mapping for R4

### DISCUSSION THREAD

**Discussion by phorusplay** [WLAN-Pi/feedback#42] [Hardware - WLAN Pi R4] [answered]
I am quite new to this community and want to express my appreciation for all the great work the people here have done. 

I installed WLAN Pi OS v3.1.1 on a system I put together using the kit (WLAN Pi R4 BYOP Kit) and an RP4. The system works fine except that the front button mapping is incorrect for R4. Can anyone point me to a right direction for modifying the code myself or changing configurations? Or is there a different OS package I need to install for this hardware?

Thanks.

**Comment by jiribrejcha** [comment]
The button mapping on R4 should work out of the box with no further tweaks. 

1. Are you using the display module supplied in the BYOP kit?
2. What orientation are you using your WLAN Pi?
3. What buttons are mapped incorrectly? Just one or all of them?

**Comment by jiribrejcha** [comment]
To answer your question, line 115 has the GPIO pin to button mapping:

https://github.com/WLAN-Pi/wlanpi-fpms/blob/main/fpms/modules/constants.py

**Reply by phorusplay** [reply]
1. Yes, I am using the display module in the BYOP kit with four buttons
2. Landscape
3. I am not sure if the problem I am having is really a button mapping one. 

First, I don't have the menu screen. Instead, it shows the initial screen with the IP info. No button gets me to the menu. The button on the left does not seem to do anything other than its apparent function as 'enter' when the other three buttons are used to get to action items. The bottom button is for either SHUTDOWN or REBOOT, the middle button for switching HOTSPOT or CLASSIC, the top button gets to the 'reachabilitty'. However, once these buttons pressed, there is no way to get back to the initial screen.

Maybe I did not clearly understand how the buttons should work. 

On reviewing the code, there is no option for four buttons. Does this sound like I have a unsupported screen?

**Reply by jiribrejcha** [reply]
Is the joystick working on your display module?

**Reply by phorusplay** [reply]
I did not realize it was joystick. Everything works fine. Thank you very much.

**Comment by jiribrejcha** [comment, marked as answer]
No problem at all 👍🏻 Enjoy your WLAN Pi.
